src/icalendar/cal/component.py

```diff
@@ -663,5 +670,69 @@ def new(
         component.comments = comments
         return component
 
+    def to_jcal(self) -> list[str, list, list]:
+        """Convert this component to a jcal object.
+
+        Returns:
+            jcal object
+
+        See also :attr:`to_json`.
+        """
+        properties = []
+        for key, value in self.items():
+            for item in value if isinstance(value, list) else [value]:
+                properties.append(item.to_jcal(key.lower()))
+        return [
+            self.name.lower(),
+            properties,
+            [subcomponent.to_jcal() for subcomponent in self.subcomponents],
+        ]
+
+    def to_json(self) -> str:
+        """Return this component as a jCal JSON string.
+
+        Returns:
+            JSON string
+
+        See also :attr:`to_jcal`.
+        """
+        return json.dumps(self.to_jcal())
+
+    @classmethod
+    def from_jcal(cls, jcal: str | list) -> Component:
+        """Create a component from a jcal object.
+
+        Returns:
+            Component
+
+        Raises:
+            JCalParsingError
+
+        This reverses :attr:`to_json` and :attr:`to_jcal`.
+        """
+        if isinstance(jcal, str):
+            jcal = json.loads(jcal)
```

Comment:
Handle possible exception and convert into `JCalParsingError `?

Reply:
I decided that I rather declare the exception. Everything inside the parsed json is JCalParsingError but this way, it is easier to understand what is the problem.

What do you think?